src/api/avatar.js

```diff
@@ -0,0 +1,49 @@
+import { ethers, getNetworkId, getNetworkProviderUrl } from '@ensdomains/ui'
+
+export default async function validateTokenURI(value, addr) {
+  const details = value.split('/')
+
+  // basic validity checks
+  if (value.split('/').length !== 3) return false
+  if (details[0] !== 'eip155:1') return false
+
+  const [schema, contractAddress] = details[1].split(':')
+  const tokenId = parseInt(details[2])
+
+  // token/contract checks
+  if (schema !== 'erc721' && schema !== 'erc1155') return false
+  if (isNaN(tokenId)) return false
+
+  try {
+    const network = await getNetworkId()
+    const networkProvider = getNetworkProviderUrl(`${network}`)
+    const provider = new ethers.providers.JsonRpcProvider(networkProvider)
+    // set functions needed to check validity of ownership/NFT compatibility
+    const ABI =
+      schema === 'erc721'
+        ? [
+            'function tokenURI(uint256 tokenId) external view returns (string memory)',
+            'function ownerOf(uint256 tokenId) public view returns (address)'
+          ]
+        : [
+            'function uri(uint256 _id) public view returns (string memory)',
+            'function balanceOf(address account, uint256 id) public view returns (uint256)'
+          ]
+    const contract = new ethers.Contract(contractAddress, ABI, provider)
+
+    // if there is token metadata, and the owner of the token is the address for name, return as valid
+    if (schema === 'erc721') {
+      const tokenURI = await contract.tokenURI(tokenId)
+      const ownerOf = await contract.ownerOf(tokenId)
+      if (tokenURI && ownerOf === addr) return true
+    } else {
+      const uri = await contract.uri(tokenId)
```

Comment:
Hi @TateB, nice work! Calling one of these contract functions with a tokenId higher than 2^256 -1 throws an exception (e.g. eip155:1/erc1155:0x495f947276749ce646f68ac8c248420045cb7b5e/8112316025873927737505937898915153732580103913704334048512380490797008551937). 
![image](https://user-images.githubusercontent.com/53792888/145445694-fe6b4ca0-14aa-4343-9ea3-c4b4d71b5cb6.png)
Using a BigNumber instance for tokenId instead of a number should avoid integer overflow.
`ethers.BigNumber.from("8112316025873927737505937898915153732580103913704334048512380490797008551937")`

Reply:
#1395

Reply:
fixed :)

Reply:
good catch @arnaud-eb